docs(kotlin-script): remove metadata, rewrite (#20192)
renovatebot/renovate: lib/modules/manager/kotlin-script/readme.md

Renovate supports upgrading dependencies in [Kotlin Script](https://github.com/Kotlin/KEEP/blob/master/proposals/scripting-support.md) files.

> These are self-contained scripts where one can write Kotlin code with JVM backend, and compilation happens when the scripts are ran.

Author: ```suggestion
You can use Kotlin Script to write Kotlin code with the JVM backend.
Compilation happens when the Kotlin Script runs.
```

The current sentence feels ungrammatical. How about this? Also should it be `with the JVM backend` or `on the JVM backend`?

Reviewer: I think this sentence can be removed as it isn't really relevant to Renovate.

Reviewer 2: I agree regarding the removal, but I think we should keep the current link or a similar one e.g. https://kotlinlang.org/docs/custom-script-deps-tutorial.html

Author: ```suggestion
Read the [Kotlin Script docs](https://kotlinlang.org/docs/custom-script-deps-tutorial.html) to learn more.
```

// ...
```

By default, Renovate only scans files with the `.main.kts` extension and not `.kts`.
This way Renovate avoids ambiguity with Gradle config files that use the `.gradle.kts` extension.

> If you have Kotlin Script files not matching the default filter, you may use the `fileMatch` config option to let Renovate find such files:

Author: ```suggestion
If you want to manage other Kotlin Script files, you may use the `fileMatch` config option to let Renovate update these files:
```

> If you have Kotlin Script files with the `.kts` extension, you may use the `fileMatch` config option to let Renovate update the `.kts` files:

Author: I've dropped the text that said: _or no extension is used,_. Do we want to add a configuration example for the case where no extensions are used? We only have an example for the `.kts` case now...

Reviewer 2: ```suggestion
If you want to manage other Kotlin Script files, you may use the `fileMatch` config option to let Renovate update these files:
```
This way we include the arbitrary Kotlin script files.

> If you have Kotlin Script files with the `.kts` extension, you may use the `fileMatch` config option to let Renovate update the `.kts` files:

Reviewer: ```suggestion
If you have Kotlin Script files not matching the default filter, you may use the `fileMatch` config option to let Renovate find such files:
```
